src/util/system.cpp

```diff
@@ -399,6 +399,15 @@ std::optional<unsigned int> ArgsManager::GetArgFlags(const std::string& name) co
     return std::nullopt;
 }
 
+fs::path ArgsManager::GetNormalPath(std::string pathlike_arg) const
+{
+    const std::string path_str{GetArg(pathlike_arg, "")};
+    if (path_str.empty()) return {};
+    // Adding a slash followed by parent_path() call
+    // guaranties no trailing slashes in the resulted path.
+    return fs::PathFromString(path_str + "/").lexically_normal().parent_path();
```

Comment:
In commit "util: Add ArgsManager::GetNormalPath() function" (034f12a115e21b4205b532a3f946b129e83505d9)

I think previous approach which avoids direct string manipulation and avoids assuming "/" is a path separator is a little cleaner. Would suggest something like:

```c++
fs::path result = fs::PathFromString(path_str).lexically_normal(); // Collapse slashes and dots
return result.has_filename() ? result : result.parent_path(); // Remove trailing slash, if present
```

Also it would be great to write a simple unit test for this new function.

Reply:
Thanks! [Done](https://github.com/bitcoin/bitcoin/pull/24265#issuecomment-1030623898).